http/client.go

```diff
@@ -216,6 +217,10 @@ func (c *client) DoRequest(req *APIRequest, data interface{}) error {
 	return nil
 }
 
+func (c *client) IsAuthenticated() bool {
```

Comment:
Is `IsAuthenticated` a good naming? I'm not sure I understand what it tests, but I would expect the `IsAuthenticated` method to return true when the HTTP client is already authenticated to the API

Reply:
There is no notion of 'already authenticated' to the API, only the ability to be authenticated against the API.
This was the best naming i could think of but i'm open to better idea.

Reply:
Then I think I find this name confusing... `IsAuthenticatedClient`?

Reply:
Ok !

Reply:
Second thought is the Client necessary since the structure is already a Client ?

This would make the signature look like: `client.IsAuthenticatedClient()` isn't the Client a bit redundant ?

Reply:
Yes I understand, but without it, I find it a bit confusing: is the `Authenticated` word about the client or about a request? That's why I would prefer to explicitly add it here. Even though I understand after your explanation that there is no such thing as "already authenticated" to the API